booking-app/src/main/java/org/motechproject/bookingapp/helper/DtoLookupHelper.java

```diff
@@ -49,6 +55,59 @@ public static BookingGridSettings changeLookupForPrimeVaccinationSchedule(Bookin
         return settings;
     }
 
+    public static BookingGridSettings changeLookupForScreening(BookingGridSettings settings) throws IOException {
+        Map<String, Object> fieldsMap = new HashMap<>();
+        DateTimeFormatter formatter = DateTimeFormat.forPattern("yyyy-MM-dd");
+
+        DateFilter defaultDateFilter = DateFilter.TODAY;
+
+        if (StringUtils.isBlank(settings.getFields())) {
+            settings.setFields("{}");
+        }
+
+        if (StringUtils.isBlank(settings.getLookup())) {
+            settings.setLookup("Find By Date");
+        } else {
+            String fields = settings.getFields();
+            fieldsMap = OBJECT_MAPPER.readValue(fields, new TypeReference<HashMap>() {});
+            if (settings.getLookup().equals("Find By Volunteer Name")) {
+                settings.setLookup("Find By Date And Volunteer Name");
+            } else {
+                settings.setLookup("Find By Date And Clinic Location");
+            }
+        }
+        if (settings.getDateFilter() != null) {
+            defaultDateFilter = settings.getDateFilter();
+        }
+
+        Map<String, String> dateRange = new HashMap<>();
+        if (defaultDateFilter == DateFilter.DATE_RANGE) {
+            if (StringUtils.isNotBlank(settings.getStartDate())) {
+                dateRange.put("min", settings.getStartDate());
+            } else {
+                dateRange.put("min", "");
+            }
+            if (StringUtils.isNotBlank(settings.getEndDate())) {
+                dateRange.put("max", settings.getEndDate());
+            } else {
+                dateRange.put("max", "");
+            }
+        } else if(defaultDateFilter == DateFilter.TODAY) {
```

Comment:
you don't need those conditions, use date range form filter filed like it was done in the ExportController